File: desktop/src/main/java/imo/ImoDialog.java
package imo;

import java.awt.event.KeyEvent;
import java.awt.event.KeyListener;

import javax.inject.Inject;
import javax.swing.JDialog;

import mine.awt.MineCanvasAWT;
import mine.awt.MineAwtUtils;
import mine.event.MineCanvas;
import mine.event.SleepManager;
import mine.paint.MineImageLoader;

public class ImoDialog extends JDialog implements KeyListener, ImoListener {

	@Inject MineCanvas mc;
	@Inject SleepManager sleepManager;
	@Inject MineImageLoader imageLoader;
	@Inject ImoCanvas ic;

	private boolean life;
	private int exp;
	private int level = 1;

	private MineCanvasAWT c;

	@Inject
	public ImoDialog() {
		super();
	}

	public void launch() {

		c = new MineCanvasAWT(mc);
		MineAwtUtils.setSize(c, 300, 300);

		ic.setup("僕", 1);

		ic.setVisible(true);
		setTitle("ImoBattle");

		c.addKeyListener(this);

		ic.setImoListener(this);

		getContentPane().add(c);
		pack();
		MineAwtUtils.setCenter(this);
		setVisible(true);
		c.requestFocus();
	}



	public void gameExit(int exp_) {
		this.life = true;
		this.exp = exp_;
		if ( exp > 0) level++;
		ic.gameReset("俺", level);
	}

	public int getExp() {
		return exp;
	}

	public void setExp(int exp) {
		this.exp = exp;
	}

	public int getLevel() {
		return level;
	}

	public void setLevel(int level) {
		this.level = level;
	}

	public boolean isLife() {
		return life;
	}

	public void setLife(boolean life) {
		this.life = life;
	}

	@Override
	public void keyReleased(KeyEvent e) {
		ic.keyReleased(e.getKeyChar(), e.getKeyCode());
	}

	@Override
	public void keyPressed(KeyEvent e) {
		ic.keyPressed(e.getKeyChar(), e.getKeyCode());
	}

	@Override
	public void keyTyped(KeyEvent e){
	}

}

File: desktop/src/main/java/imo/ImoDialog.kt
package imo

import java.awt.event.KeyEvent
import java.awt.event.KeyListener

import javax.inject.Inject
import javax.swing.JDialog

import mine.awt.MineCanvasAWT
import mine.awt.MineAwtUtils
import mine.event.MineCanvas
import mine.event.SleepManager
import mine.paint.MineImageLoader
import javax.inject.Singleton
import kotlin.concurrent.thread

@Singleton
class ImoDialog
@Inject
constructor() : JDialog(), KeyListener, ImoListener {

    @field: Inject
    lateinit internal var mc: MineCanvas

    @field: Inject
    lateinit internal var sleepManager: SleepManager

    @field: Inject
    lateinit internal var imageLoader: MineImageLoader

    @field: Inject
    lateinit internal var ic: ImoCanvas

    var life: Boolean = false
    var exp: Int = 0
    var level = 1

    fun launch() {

        val c = MineCanvasAWT(mc)
        MineAwtUtils.setSize(c, 300, 300)

        ic.setup("僕", 1)

        ic.setVisible(true)
        title = "ImoBattle"

        c.addKeyListener(this)

        ic.setImoListener(this)

        contentPane.add(c)
        pack()
        MineAwtUtils.setCenter(this)
        isVisible = true
        c.requestFocus()

        thread {
            while (true) {
                if (mc.isUpdated) {
                    c.repaint()
                }
                Thread.sleep(10)
            }
        }
    }


    override fun gameExit(exp_: Int) {
        this.life = true
        this.exp = exp_
        if (exp > 0) level++
        ic.gameReset("俺", level)
    }

    override fun keyReleased(e: KeyEvent) {
        ic.keyReleased(e.keyChar, e.keyCode)
    }

    override fun keyPressed(e: KeyEvent) {
        ic.keyPressed(e.keyChar, e.keyCode)
    }

    override fun keyTyped(e: KeyEvent) {
    }

}
